cpp/bench/sg/main.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_runtime.h>
+#include <utils.h>
+#include <chrono>
+#include "harness.h"
+
+int main(int argc, char **argv) {
+  try {
+    using namespace ML::Bench;
+    auto start = std::chrono::high_resolution_clock::now();
+    cudaFree(nullptr);  // just to create context!
+    Harness::Init(argc, argv);
+    Harness::RunAll();
+    Harness::PrintResultsInCsv();
```

Comment:
Ah, I like this- I was going to ask if there were any thoughts about writing a CSV outputter, but stdout should be good enough to route to a file.

Reply:
This is no longer an issue as I have just updated the interface to work with google bench.